### PR TITLE
Fix #3525 - Profiler breaks on Postgres data

### LINKED ISSUE #3525: Profiler breaks on Postgres data
**Affected module**
Ingestion

**Describe the bug**
1. Invalid length function

image

2. 
image

3.
 image

Reported by @kevinbtai

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

This PR fixes #3525.

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/orm_profiler/orm/functions/length.py [hunk 1]
```diff
@@ -33,10 +31,9 @@ def _(element, compiler, **kw):
...
-@compiles(LenFn, DatabaseServiceType.SQLite.value.lower())
-@compiles(LenFn, DatabaseServiceType.Vertica.value.lower())
-@compiles(
-    LenFn, DatabaseServiceType.Hive.value.lower().encode()
-)  # For some reason hive's dialect is in bytes...
+@compiles(LenFn, Dialects.SQLite)
+@compiles(LenFn, Dialects.Vertica)
+@compiles(LenFn, Dialects.Hive)  # For some reason hive's dialect is in bytes...
+@compiles(LenFn, Dialects.Postgres)
 def _(element, compiler, **kw):
```
Comment (pmbrull) [inline ingestion/src/metadata/orm_profiler/orm/functions/length.py:38]: Added postgres for the Length function

File: ingestion/src/metadata/orm_profiler/orm/registry.py [hunk 2]
```diff
@@ -27,13 +27,39 @@ class CustomTypes(TypeRegistry):
     UUID = UUIDString
 
 
+class Dialects(Enum):
+    Hive = b"hive"
+    Postgres = "postgresql"
```
Comment (pmbrull) [inline ingestion/src/metadata/orm_profiler/orm/registry.py:32]: Created a new Enum where we can easily list down the schemes for each source. We base each entry in the Data Types for the services and the value we can tune it depending on the scheme. This simplifies the usage and allows for non-direct handling of the dialect names as with Postgres

File: ingestion/src/metadata/utils/column_type_parser.py [hunk 3]
```diff
@@ -70,6 +70,7 @@ class ColumnTypeParser:
         "DATETIMEOFFSET": "DATETIME",
         "DECIMAL": "DECIMAL",
         "DOUBLE PRECISION": "DOUBLE",
+        "DOUBLE_PRECISION": "DOUBLE",
```
Comment (pmbrull) [inline ingestion/src/metadata/utils/column_type_parser.py:73]: Missing type in postgres

File: ingestion/src/metadata/utils/column_type_parser.py [hunk 4]
```diff
@@ -101,7 +102,6 @@ class ColumnTypeParser:
         "INTERVAL DAY TO SECOND": "INTERVAL",
         "INTERVAL YEAR TO MONTH": "INTERVAL",
         "INTERVAL": "INTERVAL",
-        "JSON": "JSON",
```
Comment (pmbrull) [inline ingestion/src/metadata/utils/column_type_parser.py:104]: JSON entry was repeated in the mapping

**Review by harshach — APPROVED** [review]

**Pull request merged**